### PR TITLE
Refactor styling and overflow, fix like button on details page

### PR DESCRIPTION
**Pull request by ncbecker**

### CODE REVIEW

File: client/src/components/IconButton.js [hunk 1]
```diff
@@ -17,6 +18,7 @@ export const IconButton = styled.button`
   align-items: center;
   width: ${(props) => (props.nav ? "48px" : "auto")};
   height: ${(props) => (props.nav ? "48px" : "auto")};
```
Comment (jamarob) [inline client/src/components/IconButton.js]: When you set multiple rules you can do it like this
```suggestion
import {css} from 'styled-components'
//...

${(props) => props.nav 
  ? css`
    width: 48px;
    height: 48px;
  `
  : css`
    width: auto;
    height: auto;
  `
`}
```

File: client/src/pages/EpisodeDetailsPage.js [hunk 2]
```diff
@@ -12,18 +12,22 @@ import { ReactComponent as NotLiked } from "../assets/icon-heart-empty.svg";
...
 } from "../utils/api.js";
 
 const PageWrapper = styled.div`
   width: 100%;
   height: 100vh;
   display: grid;
   grid-template-rows: repeat(6, auto) 1fr;
+  overflow-y: auto;
+  ::-webkit-scrollbar {
+    width: 0;
```
Comment (jamarob) [inline client/src/pages/EpisodeDetailsPage.js:25]: Do you have another scroll indicator for the user? Maybe rethink this.

File: client/src/pages/FilterPage.js [hunk 3]
```diff
@@ -13,7 +13,10 @@ const FilterPageWrapper = styled.div`
   background: ${(props) => props.theme.inputOverlay};
   border-radius: 15px 15px 0 0;
   transition: height 0.4s ease-in-out;
-  overflow: hidden;
+  overflow: scroll;
+  ::-webkit-scrollbar {
+    width: 0;
```
Comment (jamarob) [inline client/src/pages/FilterPage.js:18]: Not sure, see comment above

File: client/src/utils/api.js [hunk 4]
```diff
@@ -82,6 +82,16 @@ export async function getSingleEpisodeFromDB(id) {
   return singleEpisode;
 }
 
+export async function addOrUpdateEpisodeInDB(id, userId, liked) {
```
Comment (jamarob) [inline client/src/utils/api.js:85]: A POST should create a new entity and not update an existing one, this breaks http semantics. PATCH and PUT are idempotent and more suited for updating.

File: server.js [hunk 5]
```diff
@@ -279,10 +279,20 @@ app.get("/api/db/users", async (request, response) => {
...
+      if (liked) {
+        await updateEpisodeLiked(id, userId);
+      } else {
+        await updateEpisodeUnliked(id, userId);
+      }
+      response.status(200).send("Database updated.");
+    } else {
+      await setEpisode(id, userId);
+      response.status(200).send("Episode is set in database.");
+    }
```
Comment (jamarob) [inline server.js:295]: Better split this into two routes, one for creation and one for updating. See HTTP method specs https://tools.ietf.org/html/rfc7231#section-4.3

**Review by jamarob — COMMENTED** [review]
Like a sir 🧐

**Comment by ncbecker** [comment]
@jamarob thanks for your review and comments!

You definitely found two pain points, which are not ideally solved yet: overflow handling and my server routes. I will address both of your comments in separate branches with a more extensive refactoring.

All other suggestions will be implemented before merging 👍

**Pull request merged**